tests/testthat/test-assertions.R

```diff
@@ -70,3 +81,17 @@ test_that("validate_rows throws a message if there are no columns selected", {
 
   expect_message(validate_rows(validation, rowSums, function(x) TRUE))
 })
+
+test_that("validation returns assert_success or assert_errors attribute based on result", {
+  data <- data.frame(
```

Comment:
It seems that those test cases would benefit from the `fixtures` concept. Basically you can prepare a function that will build those data.frames for you, with the given context. You can think about the [fixtuRes](https://github.com/jakubnowicki/fixtuRes) package.
Why it is beneficial? It will make explicit what you want to achieve on the general level (here: V1 with not-negative numbers, V2 with negative numbers) as well as make the tests randomised.